walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,117 @@
+    if (sortedUnavailableTimes == null) {
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    ArrayList<TimeRange> distinctUnavailableTimes = getDistinctUnavailableTimes(sortedUnavailableTimes);
+    ArrayList<TimeRange> availableTimes = getFilteredAvailableTimes(distinctUnavailableTimes, request.getDuration());
+    return availableTimes;
+  }
+  
+  /** Helper function that returns a ArrayList of all the time ranges that the group of attendees is unavailable for, sorted by start time. **/
+  private ArrayList<TimeRange> getSortedUnavailableTimes(Collection<String> attendees, Collection<Event> events) {
+    ArrayList<TimeRange> unavailableTimes = new ArrayList<>();
+    for (Event e : events) {
+      for (String a : e.getAttendees()) {
+        // Check if any of the requested attendees are also attending the event.
+        if (attendees.contains(a)) {
+          // If one attendee is attending the event, add the time range as unavailable and end the loop.
+          unavailableTimes.add(e.getWhen());
+          break;
+        }
+      }
+    }
+
+    // Check edge case where all the events provided don't have one of the given attendees
+    if (unavailableTimes.size() == 0) {
+      return null;
+    }
+    // Sort all of the unavailable time slots by start time.
+    Collections.sort(unavailableTimes, TimeRange.ORDER_BY_START);
+    return unavailableTimes;
+  }
+
+  /** Helper function that returns all the distinct unavailable times by merging times that are equal or overlapping. **/
+  private ArrayList<TimeRange> getDistinctUnavailableTimes(ArrayList<TimeRange> sortedUnavailableTimes) {
+    ArrayList<TimeRange> distinctUnavailableTimes = new ArrayList<>();
+    TimeRange mergedTime = sortedUnavailableTimes.get(0);
+    for (int i = 1; i < sortedUnavailableTimes.size(); i++) {
+      TimeRange time = sortedUnavailableTimes.get(i);
+      // If the current time overlaps with the previous time, merges them.
+      if (mergedTime.overlaps(time)) {
+        mergedTime = mergeEventTimes(mergedTime, time);
+      } else {
+        // If current time doesn't overlap, adds the previous time to the ArrayList. Then, sets up the current time to check merge.
+        distinctUnavailableTimes.add(mergedTime);
+        mergedTime = time;
+      }
+    }
+
+    // Add the last element, if it was merged.
+    distinctUnavailableTimes.add(mergedTime);
+    return distinctUnavailableTimes;
+  }
+  
+  /** Helper function that merges two time ranges and returns a new, longer (or equal) time range. **/
+  private TimeRange mergeEventTimes(TimeRange time1, TimeRange time2) {
+    int newStart = Math.min(time1.start(), time2.start());
+    int newEnd = Math.max(time1.end(), time2.end());
+    TimeRange newTimeRange = TimeRange.fromStartEnd(newStart, newEnd, false);
+    return newTimeRange;
+  }
+
+  /** Helper function that returns the available times for all attendees that are at least as long as the duration of the meeting. **/
```

Comment:
nit: For java code, It's good practice to keep no more than 80 characters in one line.

Reply:
fixed by limiting each line to a max of 80 characters